local-cluster/src/local_cluster.rs

```diff
@@ -312,7 +312,10 @@ impl LocalCluster {
             .native_instruction_processors
             .extend_from_slice(&config.native_instruction_processors);
 
-        let (leader_ledger_path, _blockhash) = create_new_tmp_ledger!(&genesis_config);
+        let (leader_ledger_path, _blockhash) = create_new_tmp_ledger_with_size!(
+            &genesis_config,
+            config.validator_configs[0].max_genesis_archive_unpacked_size,
```

Comment:
I would have probably asked the same question :)
But it seems that this function has a bunch of assumptions about the number of validator configurations need to be present.
Just two lines below, you can see:

https://github.com/anza-xyz/agave/blob/7dffe729cd80e17789ac7de8fe4cc9d778268d63/local-cluster/src/local_cluster.rs#L320

And there are a number of `assert_eq!` checks, like this one:
https://github.com/anza-xyz/agave/blob/7dffe729cd80e17789ac7de8fe4cc9d778268d63/local-cluster/src/local_cluster.rs#L235
and then just a few lines below:
https://github.com/anza-xyz/agave/blob/7dffe729cd80e17789ac7de8fe4cc9d778268d63/local-cluster/src/local_cluster.rs#L256

I do not fully understand the logic here, but it does seem a bit fragile.
I think we often use the first validator configuration for some parameters of the whole cluster.
So, my guess is that a proper solution might be to add an invariant for the entire function, such as "there should be at least one validator in a cluster".
Check this invariant at the beginning of this function with a nice error message.
And then assign this configuration into a local variable that is responsible for the cluster configuration.  Rather than reaching for `something[0]` all the time.

But it is a larger change, so, if I want to follow the current function style, I think, `[0]` is the right call.

I've moved the `leader_config` assignment a few lines up and used it here, to reduce the number of `[0]` calls.

Reply:
Yeah, I like that idea (single check at the beginning and maybe extract out reference to the first config that can be used later). Can address in later PR